### DISCUSSION TITLE
random chat messages read for more info

### DISCUSSION THREAD

**Discussion by tk746** [PrismarineJS/mineflayer#1959] [Q&A] [answered]
is there a way to randomize chat messages so like when my bot says hi i have two messages 
1. hey my friend "username" joined

or 

2. hey welcome to the part "username" 
these are just examples but is there a way to make the bot randomly say these but garneted to say one or the other or if there's three on of them?

**Comment by aesthetic0001** [comment, marked as answer]
Elaborate. Do you mean two possible formats that it may use to respond? if that's the case,  I guess a way could be to store them in a constant array with (possibly an object) info on where the username should be inserted and what the chat message is, then just use 
```js
Math.floor(Math.random * arrayName.length - 1)
```
 to get a random index when u want to send a message and then send that message with the given info from the object u got from the index of `arrayName`
Hope this makes sense xD

**Reply by tk746** [reply]
Well i figured some of it out but when i try it it says welcome to the server "Undefinded"


heres the code

var items = ['hi, ' + bot.player, 'welcome to the server ' + bot.player];
var item = items[Math.floor(Math.random() * items.length)];

bot.on('playerJoined', (player) => {
  if (player.username !== bot.username) {
    bot.chat(item)
  }
})

**Reply by amoraschi** [reply]
Move the Array and Math.floor thing inside the event, also bot.player is an object indicating the player of the Bot, not the player that joined.

**Reply by satyamedh** [reply]
OP mark this as answer please

@tk7462
